src/fmgc/src/guidance/vnav/DescentProfile.ts

```diff
@@ -0,0 +1,175 @@
+
+        for (const leg of legs) {
+            // iterate over legs to get alt & speed constraints
+            // Constraint applies to the terminator of each leg (dest waypoint)
+            // Process conflicting constraints as well
+        }
+
+        this.recomputeIdle(meteringFix);
+    }
+
+    recomputeIdle(meteringFix: Leg): void {
+        const meteringFixAltitude = 15000; // ! placeholder
+        const meteringFixDistanceAlongFP = 25; // ! placeholder
+
+        const initialDistanceFromTOD = ((this.cruiseAltitude - meteringFixAltitude) / 100) / 3.0;
+
+        do {
+            // // First step
+            // const firstStepResult = Predictions.altitudeStep(
+            //     this.initialAltitude,
+            //     firstStepSize,
+            //     this.climbCAS,
+            //     this.climbMach,
+            //     climbN1,
+            //     this.zeroFuelWeight,
+            //     this.fuelWeight,
+            //     0,
+            //     this.isaDeviation,
+            //     this.tropoAltitude,
+            // );
+
+            // let predDistanceTraveled = firstStepResult.distanceTraveled;
+            // let predTimeElapsed = firstStepResult.timeElapsed;
+            // let predFuelWeight = this.fuelWeight - firstStepResult.fuelBurned;
+
+            // // Loop (ignoring constraints for now)
+            // // TODO: stop at each waypoint as well as SPD LIM, and record time elapsed and fuel weight
+            // for (let alt = this.cruiseAltitude - firstStepSize; alt > finalStepAltitude; alt -= 1000) {
+            //     const stepResult = Predictions.altitudeStep(
+            //         alt,
+            //         1000,
+            //         this.climbCAS,
+            //         this.climbMach,
+            //         climbN1,
+            //         this.zeroFuelWeight,
+            //         predFuelWeight,
+            //         0,
+            //         this.isaDeviation,
+            //         this.tropoAltitude,
+            //     );
+            //     predDistanceTraveled += stepResult.distanceTraveled;
+            //     predTimeElapsed += stepResult.timeElapsed;
+            //     predFuelWeight -= stepResult.fuelBurned;
+            // }
+
+            // const finalStepResult = Predictions.altitudeStep(
+            //     finalStepAltitude,
+            //     finalStepSize,;
+            //     this.climbCAS,
+            //     this.climbMach,
+            //     climbN1,
+            //     this.zeroFuelWeight,
+            //     predFuelWeight,
+            //     0,
+            //     this.isaDeviation,
+            //     this.tropoAltitude,
+            // );
+            // predDistanceTraveled += finalStepResult.distanceTraveled;
+            // predTimeElapsed += finalStepResult.timeElapsed;
+            // predFuelWeight -= finalStepResult.fuelBurned;
+
+            // this.distanceFromPposToTopOfClimb = predDistanceTraveled;
+        } while (true);
```

Comment:
This seems dangerous to exist... I guess luckily nothing calls it yet.

Reply:
removed